sdk/src/main/java/com/google/cloud/dataflow/sdk/io/TextIO.java

```diff
@@ -471,7 +471,31 @@ private Read() {}
       return new Bound<>(DEFAULT_TEXT_CODER).withoutValidation();
     }
 
-    // TODO: appendingNewlines, header, footer, etc.
+    /**
+     * Returns a transform for writing to text files that adds a header string to the files
+     * it writes.
+     *
+     * <p>A {@code null} value will clear any previously configured header.
+     *
+     * @param header the string to be added as file header
```

Comment:
javadoc should specify that a newline will be unconditionally added after the header and footer.